### ISSUE TITLE
Doc: add better scenario-based tutorial in doc

### ISSUE THREAD

**Issue by Banyc** [microsoft/reverse-proxy#751]
### What should we add or change to make your life better?

Add better scenario-based tutorial in doc.

The scenario, which has a frontend and a backend app on the same machine, is a more common case to start with.

Take an example from [Vue doc](https://cli.vuejs.org/config/#devserver-proxy):

```js
module.exports = {
  devServer: {
    proxy: {
      '^/api': {
        target: '<url>',
        ws: true,
        changeOrigin: true
      },
      '^/foo': {
        target: '<other_url>'
      }
    }
  }
}
```

... which is straightforward.

### Why is this important to you?

It will be easier to understand and follow.

**Comment by Banyc** [comment]
I have tried the following config in `appsettings.json` from the frontend:

```json
"apiUrl": "http://localhost:5000/api/",
  "Kestrel": {
    "EndPoints": {
      "Http": {
        "Url": "http://*:5000"
      }
    }
  },
  "ReverseProxy": {
    "Routes": [
      {
        "RouteId": "route1",
        "BackendId": "backend1",
        "Match": {
          "Path": "/api/"
        },
        "Transform": [
					{ "Transformation": "StripPrefix", "Prefix": "/api" },
				]
      }
    ],
    "Backends": {
      "backend1": {
        "Destinations": {
          "backend1/destination1": {
            "Address": "http://localhost:4000/"
          }
        }
      }
    }
  }
```

## Expected behavior

The expected behavior is that when POSTing `http://localhost:5000/api/users/authenticate`, the frontend server on port 5000 should transform the path to `/users/authenticate` and forward it to the backend on port 4000. In other words, the frontend server should POSTing ``http://localhost:4000/users/authenticate`.

## Actual behavior

However, when I check the traffic by Wireshark, it turned out to be this:

A random browser port POSTing `/api/users/authenticate` to port 4000 and another random browser port POSTing `/api/users/authenticate` to port 5000.

## Other efforts

I have surfed the documentation but no clear breakdown is on this scenario.

**Comment by Banyc** [comment]
## Update

It seems that the tutorial here <https://swimburger.net/blog/dotnet/use-yarp-to-host-client-and-api-server-on-a-single-origin> covers the need!

**Comment by Tratcher** [comment]
What docs were you following when generating your original config?

It looks like you had issues getting the Match Path and Transforms set up correctly:
```
        "Match": {
          "Path": "/api/{**catch-all}"
        },
        "Transforms": [
          {
            "PathRemovePrefix": "/api"
          }
        ]
```

Side note: `{ "Transformation": "StripPrefix", "Prefix": "/api" }` should have produced a config error on startup since that's not a recognized transform.

**Comment by Banyc** [comment]
> What docs were you following when generating your original config?
> 
> It looks like you had issues getting the Match Path and Transforms set up correctly:
> 
> ```
>         "Match": {
>           "Path": "/api/{**catch-all}"
>         },
>         "Transforms": [
>           {
>             "PathRemovePrefix": "/api"
>           }
>         ]
> ```
> 
> Side note: `{ "Transformation": "StripPrefix", "Prefix": "/api" }` should have produced a config error on startup since that's not a recognized transform.

Thanks a lot! The original config is from some random google search result.

**Comment by yanmayuekong** [comment]
Configuration Files seems to require a detailed configuration parameter documentation。

**Comment by Tratcher** [comment]
See https://github.com/microsoft/reverse-proxy/pull/831

**Comment by karelz** [comment]
Triage: It should be resolved by PR #833
